--- GMSHPLUGIN/doc/salome/gui/GMSHPLUGIN/CMakeLists.txt.org	2025-02-26 10:46:07.046044834 +0100
+++ GMSHPLUGIN/doc/salome/gui/GMSHPLUGIN/CMakeLists.txt	2025-02-26 10:46:17.098139998 +0100
@@ -24,7 +24,7 @@
 SALOME_CONFIGURE_FILE(static/header.html.in ${CMAKE_CURRENT_BINARY_DIR}/static/header.html)
 SALOME_CONFIGURE_FILE(static/header_py.html.in ${CMAKE_CURRENT_BINARY_DIR}/static/header_py.html)
 
-SET(smesh_file "$(SMESH_ROOT_DIR)/bin/salome/collect_mesh_methods.py")
+SET(smesh_file "${SMESH_ROOT_DIR}/bin/salome/collect_mesh_methods.py")
 SET(kernel_file "$ENV{KERNEL_ROOT_DIR}/bin/salome/prepare_generating_doc.py")
 
 SET(ADD_VAR)
