--- netgen/libsrc/core/version.hpp.org	2025-03-25 18:39:35.840585100 +0100
+++ netgen/libsrc/core/version.hpp	2025-03-25 18:40:34.553057600 +0100
@@ -41,7 +41,14 @@
               if(!vstring.empty())
                 {
                   dot = vstring.find('-');
-                  patch = std::stoi(vstring.substr(0,dot));
+                  // Catch exception or you will find a very hard to debug
+                  // segmentation fault...
+                  try {
+                    patch = std::stoi(vstring.substr(0,dot));
+                  }
+                  catch (const std::exception& e) {
+                    patch = 0;
+                  }
                   if(dot == size_t(-1)) vstring = "";
                   else vstring = vstring.substr(dot+1, vstring.size()-dot-1);
                   if(!vstring.empty())
