--- ParaView/VTK/CMake/FindLibXml2.cmake.org	2025-02-19 14:12:54.641488000 +0100
+++ ParaView/VTK/CMake/FindLibXml2.cmake	2025-02-19 14:15:09.487531000 +0100
@@ -76,7 +76,12 @@
    HINTS
    ${PC_LIBXML_LIBDIR}
    ${PC_LIBXML_LIBRARY_DIRS}
+   NO_CACHE
    )
+# New in CMake version 3.21. If LibXml2 was found before, then the find_library
+# clears the found LIBXML2_LIBRARY variable, failing the
+# FIND_PACKAGE_HANDLE_STANDARD_ARGS
+
 
 find_program(LIBXML2_XMLLINT_EXECUTABLE xmllint)
 # for backwards compat. with KDE 4.0.x:
