--- Python/Lib/ctypes/__init__.py.org	2025-03-24 18:31:06.615031000 +0100
+++ Python/Lib/ctypes/__init__.py	2025-03-24 18:33:01.770473800 +0100
@@ -380,7 +380,10 @@
                 if '/' in name or '\\' in name:
                     self._name = nt._getfullpathname(self._name)
                     mode |= nt._LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR
-
+                # Reverting decision for Python 3.8+ to limit the search for
+                # DLLs as it pretty much messes everything up...
+                LOAD_WITH_ALTERED_SEARCH_PATH = 0x00000008
+                mode = LOAD_WITH_ALTERED_SEARCH_PATH
         class _FuncPtr(_CFuncPtr):
             _flags_ = flags
             _restype_ = self._func_restype_
--- Python/Python/dynload_win.c.org	2025-03-24 18:29:28.630944600 +0100
+++ Python/Python/dynload_win.c	2025-03-24 18:51:43.027492000 +0100
@@ -226,14 +226,8 @@
         old_mode = SetErrorMode(SEM_FAILCRITICALERRORS);
 #endif
 
-        /* bpo-36085: We use LoadLibraryEx with restricted search paths
-           to avoid DLL preloading attacks and enable use of the
-           AddDllDirectory function. We add SEARCH_DLL_LOAD_DIR to
-           ensure DLLs adjacent to the PYD are preferred. */
         Py_BEGIN_ALLOW_THREADS
-        hDLL = LoadLibraryExW(wpathname, NULL,
-                              LOAD_LIBRARY_SEARCH_DEFAULT_DIRS |
-                              LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR);
+        hDLL = LoadLibraryExW(wpathname, NULL,LOAD_WITH_ALTERED_SEARCH_PATH);
         Py_END_ALLOW_THREADS
         PyMem_Free(wpathname);
 
