--- rkCommon/cmake/rkcommon_macros.cmake.old	2024-11-02 23:10:04.655538194 +0100
+++ rkCommon/cmake/rkcommon_macros.cmake	2024-11-02 23:10:09.203550014 +0100
@@ -177,7 +177,7 @@
     endif()
 
     # Try getting TBB via config first
-    find_package(TBB 2021.1 QUIET COMPONENTS tbb tbbmalloc CONFIG)
+    find_package(TBB 2020.3 QUIET COMPONENTS tbb tbbmalloc CONFIG)
     if (TBB_FOUND)
       list(APPEND RKCOMMON_TASKING_LIBS TBB::tbb TBB::tbbmalloc)
       set(RKCOMMON_TASKING_DEFINITIONS RKCOMMON_TASKING_TBB)
